src/openrct2/paint/tile_element/Paint.TileElement.cpp

```diff
@@ -310,9 +310,9 @@ static void PaintTileElementBase(PaintSession& session, const CoordsXY& origCoor
         { 1, 7, 3 },
     };
 
-    for (int32_t sy = 0; sy < 3; sy++)
+    for (std::size_t sy = 0; sy < std::size(segmentPositions); sy++)
     {
-        for (int32_t sx = 0; sx < 3; sx++)
+        for (std::size_t sx = 0; sx < std::size(*segmentPositions); sx++)
```

Comment:
Yes, this is intentional. `segmentPositions` is a 2-dimensional array.
`std::size(segmentPositions)` becomes `std::size(int32_t arr[][3])`
`std::size(*segmentPositions)` becomes `std::size(int32_t arr[3])`

When I made this change, I've verified that they refer to the correct dimension of the array by changing the size.

Reply:
Not sure if a comment would help much, it's pretty standard C++. Instead of dereferencing, I guess the array accessor could be used:

```suggestion
        for (std::size_t sx = 0; sx < std::size(segmentPositions[sy]); sx++)
```

I think this is still `constexpr` then.

Reply:
That would be even better :+1: